server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -103,20 +103,22 @@ CassandraSessionConfiguration getCassandraSessionConfiguration(PropertiesProvide
 
     @Provides
     @Singleton
-    FetchBatchSizes getFetchBatchSizesConfiguration(PropertiesProvider propertiesProvider) {
+    BatchSizes getBatchSizesConfiguration(PropertiesProvider propertiesProvider) {
         try {
             PropertiesConfiguration configuration = propertiesProvider.getConfiguration(BATCHSIZES_FILE_NAME);
-            FetchBatchSizes fetchBatchSizes = FetchBatchSizes.builder()
-                    .metadata(configuration.getInt("metadata", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
-                    .headers(configuration.getInt("headers", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
-                    .body(configuration.getInt("body", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
-                    .full(configuration.getInt("full", FetchBatchSizes.DEFAULT_FETCH_BATCH_SIZE))
+            BatchSizes batchSizes = BatchSizes.builder()
```

Comment:
I agree we should put the default value handling in the builder, but : to do that, we should have a configuration reader returning Options. Unless we put some guava Optional everywhere, it would require java 8. I guess it can wait for java 8 migration, WDYT ?